molten-cache/src/main/java/com/hotels/molten/cache/ReactiveCache.java

```diff
@@ -43,4 +45,46 @@
      * @return a completion once the operation has finished
      */
     Mono<Void> put(K key, V value);
+
+    /**
+     * Return the value stored to {@code key} if any,
+     * otherwise subscribe to the given {@code Mono}
+     * and store its return value in the cache.
+     *
+     * <pre>
+     * return Mono.fromCallable(() -&gt; expensiveServiceCall(key))
+     *   .as(serviceCallCache.withKey(key));
+     * </pre>
+     *
+     * @param key the cache key
+     * @return a function
+     */
+    default Function<Mono<V>, Mono<V>> withKey(K key) {
```

Comment:
I'm not sure if this is the right name for this method. I'd rather call it `cachedWith` or `cachingWith`. I think it reads better:
```
expensiveMono.as(someCache.cachingWith(key))
```